python/kserve/kserve/protocol/infer_type.py

```diff
@@ -791,8 +880,14 @@ def to_grpc(self) -> ModelInferResponse:
         """
         infer_outputs = []
         raw_output_contents = []
+        use_raw_outputs = False
+        # If FP16 datatype is present in the outputs use raw outputs.
+        if _contains_fp16_datatype(self):
+            use_raw_outputs = True
         for infer_output in self.outputs:
-            if isinstance(infer_output.data, numpy.ndarray):
+            if use_raw_outputs and infer_output.data:
+                infer_output.data = infer_output.as_numpy()
+            if isinstance(infer_output.data, np.ndarray):
                 infer_output.set_data_from_numpy(infer_output.data, binary_data=True)
```

Comment:
seems like we are decoding back to numpy on line 889 and then serializing to bytes again on line 891. If it is already in bytes then we can simply pass them?